src/osd/PrimaryLogPG.cc

```diff
@@ -727,6 +728,37 @@ void PrimaryLogPG::maybe_force_recovery()
     maybe_kick_recovery(soid);
 }
 
+class PGLSNameFilter : public PGLSFilter {
+  boost::regex regex;
+public:
+  int init(bufferlist::iterator &params) override
+  {
+    try {
+      string reg_exp ;
+      ::decode(reg_exp, params);
+      if (reg_exp.empty()) {
+        return -EINVAL;
+      } else {
+        regex = reg_exp;
+      }
+    } catch (buffer::error &e) {
+      return -EINVAL;
+    }
+
+    return 0;
+  }
+  ~PGLSNameFilter() override {}
```

Comment:
No need to change existing code -- but definitely no need to add one to new code where it isn't needed.